src/Util/AstHelper.php

```diff
@@ -0,0 +1,113 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Roave\BetterReflection\Util;
+
+use Closure;
+use PhpParser\Node;
+
+use Roave\BetterReflection\BetterReflection;
+use Roave\BetterReflection\Reflection\Reflection;
+use Roave\BetterReflection\Reflection\ReflectionClass;
+use Roave\BetterReflection\Reflection\ReflectionFunction;
+
+use PhpParser\Node\Expr\Closure as ClosureNode;
+use PhpParser\Node\Expr\ClosureUse as ClosureUseNode;
+use PhpParser\Node\Expr\ArrowFunction as ArrowFunctionNode;
+
+use PhpParser\NodeTraverser;
+use PhpParser\NodeVisitor\FindingVisitor;
+use PhpParser\PrettyPrinter\Standard as StandardPrettyPrinter;
+use PhpParser\PrettyPrinterAbstract;
+use RuntimeException;
+
+final class AstHelper {
+    private BetterReflection $betterReflection;
+
+    public function __construct( ?BetterReflection $betterReflection = null ) {
+        if ( !$betterReflection  ) {
+            $this->betterReflection = new BetterReflection();
+        } else {
+            $this->betterReflection = $betterReflection;
+        }
+    }
+    protected function getBetterReflection(): BetterReflection {
+        return $this->betterReflection;
+    }
+    /**
+     * 
+     * @return Node[]|null $nodes
```

Comment:
Can we be more precise about the type of the returned array? Perhaps `list<Node>`?

When could this be `null`? Can we somehow get rid of that scenario?